actionmailer/test/log_subscriber_test.rb

```diff
@@ -35,6 +41,29 @@ def test_deliver_is_notified
     BaseMailer.deliveries.clear
   end
 
+  def test_deliver_is_notified_when_exception_is_raised
+    @original_delivery_method = ActionMailer::Base.delivery_method
+    ActionMailer::Base.delivery_method = ExceptionDelivery
+
+    mail = BaseMailer.welcome
+    assert_raise Net::SMTPAuthenticationError do
+      mail.deliver_now
+    end
+    wait
```

Comment:
I'm not sure if we have an explict style guide for this, but tests are easier to read when it goes assignments, assertions, with spaces inbetween so I think this would read better as:

```ruby
mail = BaseMailer.welcome

assert_raise Net::SMTPAuthenticationError do
  mail.deliver_now
end

wait
```